File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.cpp
#include <string>
#include <vector>
#include <unordered_set>
#include <algorithm>
#include <iostream>
#include <unordered_map>
using namespace std;

unordered_set<int> us;

int solution(int k, vector<int> tangerine) {
    int answer = 0;
    
unordered_map<int, int> cnt;
    
    
    for(int i =0;i<tangerine.size();i++){
        cnt[tangerine[i]]++;
    }
    
    auto cmp = [&cnt](int a, int b){
        return cnt[a] > cnt[b] || (cnt[a] == cnt[b] && a > b);
    };
    
    sort(tangerine.begin(), tangerine.end(), cmp);
    
    for(int i =0;i<k;i++){
        //if(us.size() >= k) break;
        
        us.insert(tangerine[i]);
        //cout << tangerine[i] << " " << us.size()<<"\n";
        
    }
    
    answer = us.size();
    
    return answer;
}

File: 프로그래머스/2/138476. 귤 고르기/귤 고르기.js
function solution(k, tangerine) {
    var answer = 0;
    
    var map = new Map();
    var list = [];
    tangerine.forEach(i=>{
        if(map.has(i)){
            map.set(i,map.get(i)+1);
        }else{
            map.set(i,1);
        }
    })
    var s = new Set();
    
    for(let [key,value] of map){
        list.push([key,value]);
    }
    
    list.sort((a,b)=> b[1] - a[1])
    
    console.log(list)
    
    for(let i = 0;i<list.length;i++){
        if(k <=0)break;
        
        k-= list[i][1]
        answer++;
    }
    
    //answer = s.size;
    
    return answer;
}